// @turners/document-generator-client/dist/esm/api.js
/* tslint:disable */
/* eslint-disable */
/**
 * Document Generator API
 * API for generating PDF documents from templates.  This microservice provides endpoints for generating various types of insurance documents in PDF format using predefined templates and data models.
 *
 * The version of the OpenAPI document: 1.0.0
 * Contact: support@autosure.co.nz
 *
 * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
 * https://openapi-generator.tech
 * Do not edit the class manually.
 */
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import globalAxios from 'axios';
// Some imports not used depending on template conditions
// @ts-ignore
import { DUMMY_BASE_URL, assertParamExists, setSearchParams, serializeDataIfNeeded, toPathString, createRequestFunction } from './common';
// @ts-ignore
import { BASE_PATH, BaseAPI, operationServerMap } from './base';
export const HealthResponseStatusEnum = {
    Healthy: 'healthy'
};
/**
 * DocumentGenerationApi - axios parameter creator
 * @export
 */
export const DocumentGenerationApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Generates a Mechanical Breakdown Insurance (MBI) quote document in PDF format using the provided quote data and the MBI quote template.
         * @summary Generate MBI Quote PDF
         * @param {MbiQuoteRequest} mbiQuoteRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        generateMbiQuote: (mbiQuoteRequest, options = {}) => __awaiter(this, void 0, void 0, function* () {
            // verify required parameter 'mbiQuoteRequest' is not null or undefined
            assertParamExists('generateMbiQuote', 'mbiQuoteRequest', mbiQuoteRequest);
            const localVarPath = `/generate/mbi-quote`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'POST' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            localVarHeaderParameter['Content-Type'] = 'application/json';
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            localVarRequestOptions.data = serializeDataIfNeeded(mbiQuoteRequest, localVarRequestOptions, configuration);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * DocumentGenerationApi - functional programming interface
 * @export
 */
export const DocumentGenerationApiFp = function (configuration) {
    const localVarAxiosParamCreator = DocumentGenerationApiAxiosParamCreator(configuration);
    return {
        /**
         * Generates a Mechanical Breakdown Insurance (MBI) quote document in PDF format using the provided quote data and the MBI quote template.
         * @summary Generate MBI Quote PDF
         * @param {MbiQuoteRequest} mbiQuoteRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        generateMbiQuote(mbiQuoteRequest, options) {
            var _a, _b, _c;
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.generateMbiQuote(mbiQuoteRequest, options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = operationServerMap['DocumentGenerationApi.generateMbiQuote']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
    };
};
/**
 * DocumentGenerationApi - factory interface
 * @export
 */
export const DocumentGenerationApiFactory = function (configuration, basePath, axios) {
    const localVarFp = DocumentGenerationApiFp(configuration);
    return {
        /**
         * Generates a Mechanical Breakdown Insurance (MBI) quote document in PDF format using the provided quote data and the MBI quote template.
         * @summary Generate MBI Quote PDF
         * @param {MbiQuoteRequest} mbiQuoteRequest
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        generateMbiQuote(mbiQuoteRequest, options) {
            return localVarFp.generateMbiQuote(mbiQuoteRequest, options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * DocumentGenerationApi - object-oriented interface
 * @export
 * @class DocumentGenerationApi
 * @extends {BaseAPI}
 */
export class DocumentGenerationApi extends BaseAPI {
    /**
     * Generates a Mechanical Breakdown Insurance (MBI) quote document in PDF format using the provided quote data and the MBI quote template.
     * @summary Generate MBI Quote PDF
     * @param {MbiQuoteRequest} mbiQuoteRequest
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof DocumentGenerationApi
     */
    generateMbiQuote(mbiQuoteRequest, options) {
        return DocumentGenerationApiFp(this.configuration).generateMbiQuote(mbiQuoteRequest, options).then((request) => request(this.axios, this.basePath));
    }
}
/**
 * HealthApi - axios parameter creator
 * @export
 */
export const HealthApiAxiosParamCreator = function (configuration) {
    return {
        /**
         * Returns the health status of the document generator service
         * @summary Health check endpoint
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        healthCheck: (options = {}) => __awaiter(this, void 0, void 0, function* () {
            const localVarPath = `/health`;
            // use dummy base URL string because the URL constructor only accepts absolute URLs.
            const localVarUrlObj = new URL(localVarPath, DUMMY_BASE_URL);
            let baseOptions;
            if (configuration) {
                baseOptions = configuration.baseOptions;
            }
            const localVarRequestOptions = Object.assign(Object.assign({ method: 'GET' }, baseOptions), options);
            const localVarHeaderParameter = {};
            const localVarQueryParameter = {};
            setSearchParams(localVarUrlObj, localVarQueryParameter);
            let headersFromBaseOptions = baseOptions && baseOptions.headers ? baseOptions.headers : {};
            localVarRequestOptions.headers = Object.assign(Object.assign(Object.assign({}, localVarHeaderParameter), headersFromBaseOptions), options.headers);
            return {
                url: toPathString(localVarUrlObj),
                options: localVarRequestOptions,
            };
        }),
    };
};
/**
 * HealthApi - functional programming interface
 * @export
 */
export const HealthApiFp = function (configuration) {
    const localVarAxiosParamCreator = HealthApiAxiosParamCreator(configuration);
    return {
        /**
         * Returns the health status of the document generator service
         * @summary Health check endpoint
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        healthCheck(options) {
            var _a, _b, _c;
            return __awaiter(this, void 0, void 0, function* () {
                const localVarAxiosArgs = yield localVarAxiosParamCreator.healthCheck(options);
                const index = (_a = configuration === null || configuration === void 0 ? void 0 : configuration.serverIndex) !== null && _a !== void 0 ? _a : 0;
                const operationBasePath = (_c = (_b = operationServerMap['HealthApi.healthCheck']) === null || _b === void 0 ? void 0 : _b[index]) === null || _c === void 0 ? void 0 : _c.url;
                return (axios, basePath) => createRequestFunction(localVarAxiosArgs, globalAxios, BASE_PATH, configuration)(axios, operationBasePath || basePath);
            });
        },
    };
};
/**
 * HealthApi - factory interface
 * @export
 */
export const HealthApiFactory = function (configuration, basePath, axios) {
    const localVarFp = HealthApiFp(configuration);
    return {
        /**
         * Returns the health status of the document generator service
         * @summary Health check endpoint
         * @param {*} [options] Override http request option.
         * @throws {RequiredError}
         */
        healthCheck(options) {
            return localVarFp.healthCheck(options).then((request) => request(axios, basePath));
        },
    };
};
/**
 * HealthApi - object-oriented interface
 * @export
 * @class HealthApi
 * @extends {BaseAPI}
 */
export class HealthApi extends BaseAPI {
    /**
     * Returns the health status of the document generator service
     * @summary Health check endpoint
     * @param {*} [options] Override http request option.
     * @throws {RequiredError}
     * @memberof HealthApi
     */
    healthCheck(options) {
        return HealthApiFp(this.configuration).healthCheck(options).then((request) => request(this.axios, this.basePath));
    }
}
